### PR TITLE
ENH: Add collect.

### PR DESCRIPTION
**Pull request by danielballan**
A nice method to have. See docstring for usage example.

### CODE REVIEW

File: streams/core.py [hunk 1]
```diff
@@ -519,3 +538,18 @@ def update(self, x, who=None):
...
+		if cache is None:
+			self.cache = deque()
+        self.cache = cache
+
+        Stream.__init__(self, child)
+
+    def update(self, x, who=None):
+        self.cache.append(x)
+
+    def flush(self, _):
```
Comment (mrocklin) [inline streams/core.py]: We should probably make `_` a keyword argument to support `s.flush()`

File: streams/core.py [hunk 2]
```diff
@@ -519,3 +538,18 @@ def update(self, x, who=None):
...
+        self.emit(self.cache)
```
Comment (mrocklin) [inline streams/core.py]: Do we want to clear out the cache?  Perhaps the following?

```python
out = tuple(self.cache())
self.cache.clear()
```

File: streams/core.py [hunk 3]
```diff
@@ -241,6 +241,25 @@ def unique(self, history=None):
...
+		>>> collector = collect(source1)
+		>>> L = collector.sink(print)
+		>>> source2.sink(collector.flush)
+		>>> source1.emit(1)
+		>>> source1.emit(2)
+		>>> source2.emit('anything')  # flushes collector
+		...
+		[1, 2]
+		"""
+		return collect(self, cache=cache)
```
Comment (mrocklin) [inline streams/core.py]: There appear to be some indentation issues here.

**Comment by danielballan** [comment]
Rebased, added tests, force-pushed.

**Comment by mrocklin** [comment]
I'll fix up flake8 issues (these were my fault) then rebase and merge.

**Comment by mrocklin** [comment]
Thanks @danielballan !

**Pull request merged**